x-pack/plugins/alerting/server/alerts_client/lib/strip_framework_fields.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  cloneDeep,
+  isArray,
+  isEmpty,
+  isFunction,
+  fromPairs,
+  isObject,
+  isPlainObject,
+  omit,
+} from 'lodash';
+import { ALERT_REASON } from '@kbn/rule-data-utils';
+import { alertFieldMap } from '@kbn/alerts-as-data-utils';
+import { RuleAlertData } from '../../types';
+
+const allowedFrameworkFields: string[] = [ALERT_REASON];
```

Comment:
`kibana.alert.reason` is defined in the `alertFieldMap` but the framework does not set it, it is returned by rule types. We may want to set it at a framework level in the future but for now, we add it to this list to allow rule types to override this value.